### PR TITLE
Implement JavaScript functions

### PR DESCRIPTION
**Pull request by DrRataplan**
TODO: XPath Functions

### CODE REVIEW

**Comment by DrRataplan** [comment]
There will be a bit of a merge conflict with the other PR I just filed. Give me a few minutes to fix them when you merge either of them.

:rocket:

**Comment by JoernT** [comment]
@DrRataplan guess this PR is missing the test cases we've created during our session?

**Comment by JoernT** [comment]
@DrRataplan but looks fine right? no conflicts. Any reasons to not merge?

**Comment by DrRataplan** [comment]
I rebased everything and I added the tests, sorry for the delay!

File: test/function.test.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+      await oneEvent(el, 'refresh-done');
+
+      const label = el.querySelector('label');
+      expect(label.innerText).to.equal('Hello World');
+    });
+  });
+
+  // The following two tests are failing. Unknown why
+  it.skip('can define a simple function', async () => {
```
Comment (DrRataplan) [inline test/function.test.js:99]: These two tests are interesting. They should work. We concluded last Tuesday on something about model binding to be delayed or something? I'm sure you know what's up.

**Pull request merged**